Neos.Flow/Classes/Security/Authentication/AuthenticationTokenResolver.php

```diff
@@ -0,0 +1,61 @@
+<?php
+namespace Neos\Flow\Security\Authentication;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\ObjectManagement\ObjectManagerInterface;
+use Neos\Flow\Security\Exception\NoAuthenticationTokenFoundException;
+
+/**
+ * The authentication token resolver. It resolves the class name of a authentication token based on names.
+ *
+ * @Flow\Scope("singleton")
+ */
+class AuthenticationTokenResolver
+{
+    /**
+     * @var ObjectManagerInterface
+     */
+    protected $objectManager;
+
+    /**
+     * Constructor.
+     *
+     * @param ObjectManagerInterface $objectManager The object manager
+     */
+    public function __construct(ObjectManagerInterface $objectManager)
+    {
+        $this->objectManager = $objectManager;
+    }
+
+    /**
+     * Resolves the class name of an authentication token. If a valid provider class name is given, it is just returned.
+     *
+     * @param string $tokenName The (short) name of the token
+     * @return string The object name of the authentication token
+     * @throws NoAuthenticationTokenFoundException
+     */
+    public function resolveTokenClass($providerName)
```

Comment:
@albe @sorenmalling 

I have questions about this...

a) the docblock seems kind of wrong
b) why no type hints?
c) why would "resolveTokenClass" give back the class name of a provider (line 49)?
d) Why woudl there be a relation between provider name and token class name at all? IMHO that is not even a 1:1 relation and I am not aware of any convention combining the two and I don't think there hsould be one.

Reply:
D'oh - indeed, that `$providerName` is supposed to be `$tokenName` - I was blind.

Reply:
ohhhh, that makes much more sense then LOL

Reply:
Nooom, I am so sorry for missing those 🙈